veps/sig-network/hotpluggable-nad-ref.md

```diff
@@ -0,0 +1,199 @@
+Currently, the whole network specification, including this reference - is immutable for the lifecycle of a VM.
+If a user wishes to change the network a VM is connected to (example: switching VLANs), they must modify the VM specification and restart the VM. 
+This proposal introduces the ability to update the NAD reference (networkName) of a network on a running VM through Live Migration. 
+This allows the host network connection to change without any reboot.
+
+## Motivation
+
+VMs often need to move between logical network segments without service interruption. Some of the scenarios include:
+- moving VMs to a new VLAN
+- TBD
+
+## Goals
+
+- Change an existing NAD reference (`spec.networks[].multus.networkName`) on a running VM.
+- Rewiring the existing host-to-pod wiring on a running VM.
+
+## Non Goals
+
+- Migrating between incompatible CNI types
+- Handling failed migration
+
+## Definition of Users
+
+- Cluster admins
+- VM owners
+
+## User Stories
+
+- As a Cluster Admin, I should be able to re-assign VMs to a new VLAN ID, without having to schedule maintenance windows for reboots or downtimes.
+- As a Cluster Admin I want to immediately move a suspicious VM from the production network to a restricted network without shutting it down, so that I can cut off lateral movement and monitor traffic without alerting the attacker and while preserving the VM’s live RAM state for forensic analysis.
+- As a Cluster Admin I want to migrate VMs from a Network Attachment Definition (NAD) backed by "Physical Uplink A" to a NAD backed by "Physical Uplink B" without service interruption, so that I can perform firmware upgrades or hardware replacements on the specific ToR switch associated with Uplink A without causing downtime for the tenants.
+- As a VM Owner of a Service provider I want to promote a tenant VM from a bandwidth-restricted network profile to an unconstrained network profile instantly, so that I can apply purchased service upgrades immediately without forcing the customer to experience a disruptive reboot which would degrade their user experience.
+- As a Cluster Admin I do not want to lose my VMs in case of a failed migration
+
+## Repos
+
+kubevirt/kubevirt
+
+## Design
+
+### Current architecture
+
+When virt-controller detects a difference between the Spec and the running status of a VM, it calculates a patch. 
+If the patch involves the networkName, it marks the change as Non-Hotpluggable. Consequently, the controller sets the RestartRequired condition on the VM. 
+The VMI remains on the old network until the user issues a restart command.
+
+### Proposed architecture
+
+Trigger migration on changing networkName rather than restart.
+
+The logic will be updated to switch to `MigrationRequired` flag from `RestartRequired` when `networkName` is updated.
+- [here](https://github.com/kubevirt/kubevirt/blob/1027ea3fd260ec0d48faa6a701ad27ba3fcfd448/pkg/network/vmliveupdate/restart.go#L112) to not add `RestartRequired`
```

Comment:
Please note that the logic adding the `RestartRequired `condition [1] is independent of the logic adding the `MigrationRequired` condition [2].

Additionally, we also need to adjust the logic that copies live-updatable fields from the VM spec to the VMI spec [3].

[1] https://github.com/kubevirt/kubevirt/blob/e661fdf24887592d74275c1c5ae9e5e1a1438433/pkg/virt-controller/watch/vm/vm.go#L2966
[2] https://github.com/kubevirt/kubevirt/blob/65460586b6a0910eb2b70098de627240a3790183/pkg/network/migration/evaluator.go#L59
[3] https://github.com/kubevirt/kubevirt/blob/7c016a3706abe6fd6627ae141ff7e04f4f0c7d81/pkg/network/controllers/vm.go#L70